fv3core/stencils/xppm.py

```diff
@@ -264,6 +272,19 @@ def __init__(self, namelist, iord, jfirst, jlast):
             domain=domain,
         )
 
+    def _get_origin_domain(
+        self, grid: GridIndexing, j_domain: Literal["compute", "full"]
+    ):
+        if j_domain == "compute":
+            add_origin = (0, 0, 0)
+            add_domain = (1, 1, 1)
+        elif j_domain == "full":
+            add_origin = (0, -grid.n_halo, 0)
+            add_domain = (1, 1 + 2 * grid.n_halo, 1)
+        origin = grid.origin_compute(add=add_origin)
+        domain = grid.domain_compute(add=add_domain)
+        return origin, domain
```

Comment:
This pattern is generic enough it would make sense to live in the grid indexing object. We do have other computations in the model that are compute in the i dimension and full in the j dimension and vice versa.

Reply:
Noting here that we discussed adding an interface like:
```python
origin, domain = grid_indexing.get_origin_domain([X_INTERFACE_DIM, Y_DIM, Z_DIM], halos=[0, grid.n_halo])
```
which I'll do in a follow-on PR.